tests/fields/test_resolvers.py

```diff
@@ -47,6 +47,25 @@ def name(self) -> str:
     assert definition.fields[0].base_resolver(None) == Query().name()
 
 
+def test_staticmethod_resolver_fields():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        @staticmethod
+        def name() -> str:
+            return "Name"
+
+    definition = Query._type_definition
+
+    assert definition.name == "Query"
+    assert len(definition.fields) == 1
+
+    assert definition.fields[0].python_name == "name"
+    assert definition.fields[0].graphql_name is None
+    assert definition.fields[0].type == str
+    assert definition.fields[0].base_resolver(None) == Query().name()
```

Comment:
should we add a test for this inside schema? so we check that the execution works fine?

Reply:
> should we add a test for this inside schema? so we check that the execution works fine?

Good idea, I added one in b1e3284757593b1bf2799f7fc1c5a5beff374893.

Reply:
It looks like this test fails on this line because of an existing bug: `StrawberryResolver.__call__` does not expect that `self.wrapped_func` can have no `self` argument.

```python
tests/fields/test_resolvers.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <strawberry.types.fields.resolver.StrawberryResolver object at 0x7fab13d045e0>, args = (None,), kwargs = {}

    def __call__(self, *args, **kwargs) -> T:
>       return self.wrapped_func(*args, **kwargs)
E       TypeError: test_staticmethod_resolver_fields.<locals>.Query.name() takes 0 positional arguments but 1 was given

strawberry/types/fields/resolver.py:39: TypeError
```

Reply:
The same error can be reproduced even without `staticmethod`:
```python
@strawberry.type
class Query:
    @strawberry.field
    def name() -> str:
        return "Name"

definition = Query._type_definition
assert definition.fields[0].base_resolver(None) == Query().name()
```

Reply:
interesting! I'll take a look at it tomorrow. /cc @BryceBeagle

Reply:
I'm trying to remember/investigate _why_ this is the case, but this is because `Query.name` != `Query().name`. I think this was something we wanted to get around to fixing at some point

Reply:
@BryceBeagle what was the resolution of this? can we merge it? :)